submitit/test_helpers.py

```diff
@@ -98,3 +98,41 @@ def submitit_file() -> Path:
     assert submitit_file() == Path(__file__).parent / "__init__.py"
     # In the job we should import submitit from the snapshot dir
     assert job.result() == tmp_path / "submitit/__init__.py"
+
+
+class FakeInfoWatcherWithTimer(core.InfoWatcher):
+    # pylint: disable=abstract-method
+    def __init__(self, delay_s: int = 60, time_change: int = 5, end_result: str = "failed"):
+        super().__init__(delay_s)
+        self.start_timer = time.time()
+        self.time_change = time_change
+        self.end_result = end_result
+
+    def get_state(self, job_id: str, mode: str = "standard") -> str:
+        now = time.time()
+        if now - self.start_timer < self.time_change:
+            return "pending"
+        elif now - self.start_timer > self.time_change and now - self.start_timer < 2 * self.time_change:
+            return "running"
+        return self.end_result
+
+
+class FakeJobWithTimer(core.Job[core.R]):
+    def __init__(self, job_id, folder, time_change: int = 5, end_result: str = "failed"):
```

Comment:
I think you could simplify the test to use the same times for both jobs, and put all the logic (including return result into the Watcher)
Anyway if I get it right we're not testing the exact time for each job, so it won't make a difference but would remove a fair amount of code.